src/components/form/text-area/index.tsx

```diff
@@ -59,8 +61,10 @@ export const TextArea = React.forwardRef<HTMLDivElement, TextAreaProps>(
     };
 
     React.useEffect(() => {
-      recalculateHeight();
-    }, []);
+      if (value && counter === 0) {
+        recalculateHeight(String(value));
+      }
+    }, [value]);
```

Comment:
Have you already tested it? With pass values on setTimeout?

Reply:
to prevent multiple useEffect calls

Reply:
Yes, i wouldn't open PR without testing. It's called only once when `value = ''`